rskj-core/src/test/java/co/rsk/net/LightProcessorTest.java

```diff
@@ -0,0 +1,412 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.net;
+
+import co.rsk.core.Coin;
+import co.rsk.core.RskAddress;
+import co.rsk.crypto.Keccak256;
+import co.rsk.db.RepositoryLocator;
+import co.rsk.db.RepositorySnapshot;
+import co.rsk.net.light.LightPeer;
+import co.rsk.net.light.LightProcessor;
+import co.rsk.net.light.message.*;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.core.*;
+import org.ethereum.crypto.HashUtil;
+import org.ethereum.db.BlockStore;
+import org.ethereum.db.TransactionInfo;
+import org.ethereum.net.MessageQueue;
+import org.ethereum.net.server.Channel;
+import org.ethereum.vm.DataWord;
+import org.ethereum.vm.LogInfo;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.ethereum.TestUtils.*;
+import static org.junit.Assert.assertArrayEquals;
+import static org.mockito.ArgumentCaptor.forClass;
+import static org.mockito.Mockito.*;
+
+/**
+ * Created by Julian Len and Sebastian Sicardi on 20/10/19.
+ */
+public class LightProcessorTest {
+
+    private static final byte[] HASH_1 = HashUtil.sha256(new byte[]{1});
+
+    private Blockchain blockchain;
+    private BlockStore blockStore;
+    private LightProcessor lightProcessor;
+    private MessageQueue msgQueue;
+    private Keccak256 blockHash;
+    private RepositoryLocator repositoryLocator;
+    private LightPeer lightPeer;
+
+    @Before
+    public void setup(){
+        blockchain = mock(Blockchain.class);
+        blockStore = mock(BlockStore.class);
+        repositoryLocator = mock(RepositoryLocator.class);
+        lightProcessor = new LightProcessor(blockchain, blockStore, repositoryLocator);
+        msgQueue = spy(MessageQueue.class);
+        blockHash = new Keccak256(HASH_1);
+        lightPeer = new LightPeer(mock(Channel.class), msgQueue);
+    }
+
+    @Test
+    public void processGetBlockReceiptMessageAndShouldReturnsReceiptsCorrectly() {
+        List<Transaction> txs = new LinkedList<>();
+        long requestId = 0;
+        List<TransactionReceipt> receipts = new LinkedList<>();
+        TransactionReceipt receipt = createReceipt();
+        receipts.add(receipt);
+        final Block block = mock(Block.class);
```

Comment:
Like ajl says on his blog https://blogs.msmvps.com/lopez/2012/08/09/tdd-unit-tests-and-mocks/ it's undesirable to mock so much when testing. Ideally, we should have factories to build these objects for us, it is a strong smell to resort to mock all these classes, especially domain classes like Block and Transaction, since it's not exactly the external, unpredictable world the one we're mocking.

There is a BlockBuilder class for this, it might be a bit unwieldy for the purposes of this test, but I think it merits trying to make efforts in order to make non mocking tests easy to code.